shared/engine/index.platform.desktop.js

```diff
@@ -11,12 +13,32 @@ class NativeTransport extends TransportShared {
     this.needsConnect = true
   }
 
-  _connect_critical_section(cb: any) {
+  _connect_critical_section = (cb: any) => {
```

Comment:
We might not want these to be arrows. These are overrides and maybe there's assumptions about 'this' we need to be careful about?

Reply:
ah, ok, i'll revert that